### PR TITLE
Added the Product Search tests and the src file updates

### PR DESCRIPTION
**Pull request by nirupa-kumar**
@nnegrey  Please review.
Added the GCS file code and the tests for Product Search

### CODE REVIEW

File: vision/product-search/cloud-client/src/main/java/com/example/vision/ProductSearch.java [hunk 1]
```diff
@@ -130,6 +128,72 @@ public static void getSimilarProducts(
...
+      System.out.println(String.format("\nProduct name: %s", product.getProduct().getName()));
+      System.out.println(
+          String.format("Product display name: %s", product.getProduct().getDisplayName()));
+      System.out.println(
+          String.format("Product description: %s", product.getProduct().getDescription()));
+      System.out.println(String.format("Score(Confidence): %s", product.getScore()));
+      System.out.println(String.format("Image name: %s", product.getImage()));
+    }
+  }
+
```
Comment (nnegrey) [inline vision/product-search/cloud-client/src/main/java/com/example/vision/ProductSearch.java]: Add the `// [END vision_product_search_get_similar_products_gcs]`

File: vision/product-search/cloud-client/src/test/java/com/example/vision/ProductSearchIT.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    String got = bout.toString();
+    assertThat(got).contains(PRODUCT_ID_1);
+    assertThat(got).doesNotContain(PRODUCT_ID_2);
+  }
+
+  @Test
+  public void testGetSimilarProductsGcsWithFilter() throws Exception {
+    // Act
+    //ImportProductSets.importProductSets(PROJECT_ID, COMPUTE_REGION, GCS_URI);
+    //bout.reset();
```
Comment (nnegrey) [inline vision/product-search/cloud-client/src/test/java/com/example/vision/ProductSearchIT.java]: Are these comments supposed to be commented or just deleted?

Comment (nirupa-kumar) [reply vision/product-search/cloud-client/src/test/java/com/example/vision/ProductSearchIT.java]: Should be deleted. Fixing that now.

**Comment by nirupa-kumar** [comment]
@nnegrey  Please review. Have fixed the issues.

**Pull request merged**